PERF_GUIDE.md

```diff
@@ -0,0 +1,76 @@
+# Build performance
+
+When we talk about build performance, it is important to understand that there
+are several build types:
+
++ cold build (booting your app up for the first time)
++ warm build (booting your app up when cache was populated)
++ rebuild (subsequent rebuilds that happen on file change)
+
+Cold build is the slowest because the cache is not yet populated and the application is booting for the first time.
+Build time varies and depends on the number of dependencies that application has but ballpark should be around 5 seconds
+for small to middle size applications and around 15 seconds for large size applications.
+
+Warm build is faster then cold one because the cache was populated already and it takes less time re-compute dependecies.
+Build time varies but ballpark should be around 2 seconds for small to middle size applications and around 10 seconds for
+large size applications.
+
+Rebuild is the fastest operantion because the cache is in prime. Build time varies but ballpark should be around 200-300ms
+for small to middle size applications and up to 1 second for large size applications.
```

Comment:
Thanks that makes sense. I was just curious about what small, middle, large meant for this guide. This gives us a good reference, for a similar sized app with 1.13.12 we're at ~11secs for a rebuild, which is way better than with 1.13.8, but still slow enough to break the dev flow. I'll dig later this week to see where the problem is. My guess is that it's something specific to what we're doing.